lib/ts/recipe/multifactorauth/index.ts

```diff
@@ -0,0 +1,148 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Recipe from "./recipe";
+import { RecipeInterface, APIOptions, APIInterface } from "./types";
+import { MultiFactorAuthClaim } from "./multiFactorAuthClaim";
+import { SessionContainerInterface } from "../session/types";
+import Multitenancy from "../multitenancy";
+import { getUser } from "../..";
+import UserMetadataRecipe from "../usermetadata/recipe";
+import { getUserContext } from "../../utils";
+
+export default class Wrapper {
+    static init = Recipe.init;
+
+    static MultiFactorAuthClaim = MultiFactorAuthClaim;
+
+    static async getFactorsSetupForUser(tenantId: string, userId: string, userContext?: Record<string, any>) {
```

Comment:
How and when is the dev supposed to use this function? If we consider the case where a user enables email-otp in the settings page, and then the dev needs to know if they should redirect the user to the email-otp setup screen or not, how will they know that? 

If the first factor is emailpassword, then we already have all the info necessary for email-otp, but yet, getFactorsSetupForUser would not return email-otp (based on our last discussion). So how is a dev supposed to know if they should redirect to the email-otp setup screen or not?

I realise that if they always redirect, and it's already setup, then our pre built UI will ask for the otp directly and not the email, but maybe this is just an unnecessary step that the dev wants to avoid (since they already have the email)

Reply:
> If the first factor is emailpassword, then we already have all the info necessary for email-otp, but yet, getFactorsSetupForUser would not return email-otp (based on our last discussion). So how is a dev supposed to know if they should redirect to the email-otp setup screen or not?

They should redirect to the email-otp screen since the factor is considered to be not set up even if we have all the information. They should not care if we have all the information or not,  only that it is not set up yet.
They do not need to set the `forceSetup` flag to true (which would make the prebuilt UI ask for the email address again).